Improve usage of `IMPORT FROM SELECT` (#19261)
pingcap/docs-cn: sql-statements/sql-statement-import-into.md

- `IMPORT INTO ... FROM SELECT` 不支持使用 `SHOW IMPORT JOB(s)` 和 `CANCEL IMPORT JOB <job-id>` 等任务管理语句。
- TiDB [临时目录](/tidb-configuration-file.md#temp-dir-从-v630-版本开始引入)需要有足够的空间来存储整个 `SELECT` 语句查询结果（暂不支持设置 `DISK_QUOTA` 选项）。
- 不支持使用 [`tidb_snapshot`](/read-historical-data.md) 导入历史数据。

> - 由于 `SELECT` 子句语法复杂，`IMPORT INTO` 的 `WITH` 参数可能与其干扰导致解析时报错。建议为复杂 `SELECT` 创建视图后，使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 导入。或者使用括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`

Reviewer: ```suggestion
- 由于 `SELECT` 子句语法复杂，`IMPORT INTO` 的 `WITH` 参数可能与其干扰导致解析时报错，比如 `GROUP BY xxx [WITH ROLLUP]` 等。建议为复杂 `SELECT` 创建视图后，使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 导入。或者使用括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`
```

Reviewer: 其实用 CTE 也行

Author: 这太高端了，我不会

Reviewer: https://github.com/pingcap/tidb/blob/3adc71cc71bd301fb38bfd510b5e39734df336fc/pkg/parser/ast/dml_test.go#L587-L588

> - 由于 `SELECT` 子句语法复杂，`IMPORT INTO` 的 `WITH` 参数可能与其干扰导致解析时报错，比如 `GROUP BY xxx [WITH ROLLUP]` 等。建议为复杂 `SELECT` 创建视图后，使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 导入。或者使用括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`

Reviewer 2: ```suggestion
- 由于 `SELECT` 子句语法复杂，`IMPORT INTO` 的 `WITH` 参数可能与其干扰导致解析时报错，比如 `GROUP BY ... [WITH ROLLUP]` 等。建议为复杂 `SELECT` 创建视图后，使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 导入。或者使用括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`
```

> - 由于 `SELECT` 子句的语法较为复杂，`IMPORT INTO` 的 `WITH` 参数可能会与其冲突，导致解析时报错，例如 `GROUP BY ... [WITH ROLLUP]` 等。建议对复杂的 `SELECT` 语句先创建视图，然后使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 进行导入。或者，可以通过括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`。

Reviewer 3: ```suggestion
- 由于 `SELECT` 子句的语法较为复杂，`IMPORT INTO` 的 `WITH` 参数可能会与其冲突，导致解析时报错，例如 `GROUP BY ... [WITH ROLLUP]`。建议先对复杂的 `SELECT` 语句创建视图，然后使用 `IMPORT INTO ... FROM SELECT * FROM view_name` 进行导入。或者，可以通过括号明确 `SELECT` 子句的范围，例如 `IMPORT INTO ... FROM (SELECT ...) WITH ...`。
```
